keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyAsyncClient.java

```diff
@@ -0,0 +1,593 @@
+     * key vault. If the named key already exists, Azure Key Vault creates a new version of the key. It requires the {@code keys/create} permission.
+     *
+     * <p>The {@code rsaKeyCreateConfig} is required and its fields {@link RSAKeyCreateConfig#name() name} and {@link RSAKeyCreateConfig#keyType() key type} cannot
+     * be null. The {@link RSAKeyCreateConfig#keySize() key size}, {@link RSAKeyCreateConfig#expires() expires} and {@link RSAKeyCreateConfig#notBefore() notBefore} values
+     * in {@code rsaKeyCreateConfig} are optional. If not specified, no values are set for the fields. The {@link RSAKeyCreateConfig#enabled() enabled} field
+     * is set to true by Azure Key Vault, if not specified.</p>
+     *
+     * <p>The {@link RSAKeyCreateConfig#keyType() keyType} indicates the type of key to create. Possible values include: {@link JsonWebKeyType#RSA RSA},
+     * {@link JsonWebKeyType#RSA_HSM RSA-HSM}.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Creates a new RSA key with size 2048 which activates in one day and expires in one year. Subscribes to the call asynchronously
+     * and prints out the newly created key details when the callback happens.</p>
+     * <pre>
+     * RSAKeyCreateConfig rsaKeyCreateConfig = new RSAKeyCreateConfig("keyName", JsonWebKeyType.RSA)
+     *    .keySize(2048)
+     *    .notBefore(OffsetDateTime.now().plusDays(1))
+     *    .expires(OffsetDateTime.now().plusYears(1));
+     *
+     * keyAsyncClient.createRSAKey(rsaKeyCreateConfig).subscribe(keyResponse -&gt;
+     *   System.out.printf("RSA Key is created with name %s and id %s \n", keyResponse.value().name(), keyResponse.value().id()));
+     * </pre>
+     *
+     * @param rsaKeyCreateConfig The key configuration object containing information about the rsa key being created.
+     * @throws NullPointerException if {@code rsaKeyCreateConfig} is {@code null}.
+     * @throws HttpRequestException if {@code name} is empty string.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value} contains the {@link Key created key}.
+     */
+    public Mono<Response<Key>> createRSAKey(RSAKeyCreateConfig rsaKeyCreateConfig) {
+        Objects.requireNonNull(rsaKeyCreateConfig, "The rsa key configuration parameter cannot be null.");
+        KeyRequestParameters parameters = new KeyRequestParameters()
+            .kty(rsaKeyCreateConfig.keyType())
+            .keySize(rsaKeyCreateConfig.keySize())
+            .keyOps(rsaKeyCreateConfig.keyOperations())
+            .keyAttributes(new KeyRequestAttributes(rsaKeyCreateConfig));
+        return service.createKey(endpoint, rsaKeyCreateConfig.name(), API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+    /**
+     * Creates a new EC key and  stores it in the key vault. The create ec key operation can be used to create any ec key type in
```

Comment:
Read through the JavaDoc and do a pass on general fix-ups, e.g. double spaces, general phrasing, etc.

Reply:
Updated